rxbindings/src/main/java/com/amplifyframework/rx/RxApiBinding.java

```diff
@@ -168,4 +173,83 @@
+     * status events.
+     * @param <T> The type representing the subscription data.
+     */
+    public static final class RxSubscriptionOperation<T> implements Cancelable {
+        private BehaviorSubject<ConnectionState> connectionStateSubject;
+        private Observable<T> subscriptionData;
+        private Cancelable amplifyOperation;
+        private String subscriptionId;
+
+        private Consumer<String> onConnected = new Consumer<String>() {
+            @Override
+            public void accept(@NonNull String subscriptionId) {
+                RxSubscriptionOperation.this.subscriptionId = subscriptionId;
+                connectionStateSubject.onNext(ConnectionState.CONNECTED);
+            }
+        };
+
+        RxSubscriptionOperation(CancelableBehaviors.StreamEmitter<String, T, ApiException> callbacks) {
+            connectionStateSubject = BehaviorSubject.create();
+            subscriptionData = Observable.create(emitter -> {
+                amplifyOperation = callbacks.streamTo(onConnected::accept,
+                                                      emitter::onNext,
+                                                      emitter::onError,
+                                                      emitter::onComplete);
+            });
+        }
+
+        /**
+         * Returns an {@link Observable} which consumers can use to
+         * retrieve data received by the subscription operation.
+         * @return Reference to the {@link Observable} with subscription data.
+         */
+        public Observable<T> observeSubscriptionData() {
+            return subscriptionData;
+        }
+
+        /**
+         * Once the subscription starts, this method returns
+         * the value of the subscriptionId.
+         * @return The value of the subscriptionId.
+         */
+        public String getSubscriptionId() {
+            return subscriptionId;
+        }
+
+        /**
+         * Returns an {@link Observable} which consumers can use to
+         * receive notfication about the status of the subscription connection. Currently,
+         * only {@link ConnectionState#CONNECTED} is emitted.
+         * @return Reference to the {@link Observable} that receives connection events.
+         */
+        public Observable<ConnectionState> observeConnectionState() {
+            return connectionStateSubject;
+        }
+
+        @Override
+        public void cancel() {
+            if (amplifyOperation != null) {
+                amplifyOperation.cancel();
+            }
+            connectionStateSubject.onComplete();
+        }
+
+        /**
+         * Enum representing connection states of a
+         * subscription operation.
+         */
+        public enum ConnectionState {
+            /**
+             * The subscription successfully established a connection and is
+             * ready to receive data.
+             */
+            CONNECTED
```

Comment:
@palpatim notes:
> FWIW, we also notify on "connecting" and "disconnected" states. "Connecting" isn't terribly meaningful, and "disconnected" degrades to a completion on the subscription data publisher, so that may not be necessary in Android

Not sure if we need to do this today -- or can defer.

Reply:
I looked into this, but it will require some additional plumbing to surface those events from the SubscriptionEndpoint class. I will make note to create a task for this.